frontend/src/routes/manage/Realm/Content/Edit/EditMode/Video.tsx

```diff
@@ -95,3 +102,99 @@ export const EditVideoBlock: React.FC<EditVideoBlockProps> = ({ block: blockRef
         </label>
     </EditModeForm>;
 };
+
+type EventSelectorProps = {
+    onChange: (eventId?: string) => void;
+    onBlur: () => void;
+    defaultValue?: Option;
+};
+
+const EventSelector: React.FC<EventSelectorProps> = ({ onChange, onBlur, defaultValue }) => {
+    const { t } = useTranslation();
+    const [error, setError] = useState(false);
+
+    const query = graphql`
+        query VideoEditModeSearchQuery($q: String!) {
+            events: searchAllEvents(query: $q) {
+                ... on EventSearchResults {
+                    items {
+                        id
+                        title
+                        seriesTitle
+                        creators
+                        thumbnail
+                        isLive
+                        created
+                        duration
+                        startTime
+                        endTime
+                    }
+                }
+            }
+        }
+    `;
+
+    const loadEvents = (input: string, callback: (options: readonly Option[]) => void) => {
+        fetchQuery<VideoEditModeSearchQuery>(environment, query, { q: input }).subscribe({
+            next: ({ events }) => {
+                if (events.items === undefined) {
+                    setError(true);
+                    return;
+                }
+
+                callback(events.items.map(item => ({
+                    ...item,
+                    // Events returned by the search API have a different ID
+                    // prefix than other events. And the mutation expects an ID
+                    // starting with `ev`.
+                    id: item.id.replace(/^es/, "ev"),
+                    syncedData: item,
+                    series: item.seriesTitle === null ? null : { title: item.seriesTitle },
+                })));
+            },
+            start: () => {},
+        });
+    };
+
+    return <>
+        {error && <Card kind="error" css={{ marginBottom: 8 }}>{t("search.unavailable")}</Card>}
+        <SearchableSelect
+            loadOptions={loadEvents}
+            format={formatOption}
+            onChange={data => onChange(data?.id)}
+            isDisabled={error}
+            {...{ onBlur, defaultValue }}
+        />
+    </>;
+};
+
+const formatOption = (event: Option, t: TFunction) => (
+    <div css={{ display: "flex", gap: 16, padding: "4px 0" }}>
+        {event.syncedData === null
+            ? <MovingTruck />
+            : <Thumbnail
+                css={{ width: 120, minWidth: 120 }}
+                event={{
+                    ...event,
+                    syncedData: {
+                        ...event.syncedData,
+                        audioOnly: false, // TODO
```

Comment:
Is this intentional?

Reply:
Yes, kind of, unfortunately. `routes/Search` has the same. Currently the search index does not yet contain information to judge whether the event is audio only because I did not want to put the whole tracks thing in the search index. I guess I should create a separate issue for that...

Reply:
I mean we can just put `audioOnly` in there, no? :shrug: But ok, whatever, ignoring it for the purposes of this PR.